### ISSUE TITLE
Animations won't "turn" unless parameters match

### ISSUE THREAD

**Issue by AdrianLxM** [faruktoptas/FancyShowCaseView#197]
## Issue part 1:
```
    var focusAnimationEnabled = true
        set(value) {
            animCounter = if (value) DEFAULT_ANIM_COUNTER else 0
            field = value
        }
```
`DEFAULT_ANIM_COUNTER` is `20`. Therefore a `focusAnimationMaxValue < 20` would lead to a situation where the animation never turns as it is already past the limit. 

##  Issue part 2:
```
            if (focusAnimationEnabled && !DISABLE_ANIMATIONS_FOR_TESTING) {
                if (animCounter == focusAnimationMaxValue) {
                    step = -1 * focusAnimationStep
                } else if (animCounter == 0) {
                    step = focusAnimationStep
                }
                animCounter += step
```
The `step` only changes sign (direction) if `focusAnimationMaxValue` or `0` is *exactly* hit. This leads to the problem in part 1. Also `focusAnimationStep` needs to be a *integer divisor* of `focusAnimationMaxValue` or the same problem occurs.

I could fix this.
Suggestion: (using `>=` and `<=` instead of `==`) and `DEFAULT_ANIM_COUNTER.coerceAtMost(focusAnimationMaxValue)`.

What branch should I base upon?

**Comment by faruktoptas** [comment]
Hi @AdrianLxM thanks a lot you catch a good point 👍 
You can base upon master branch.

**Comment by faruktoptas** [comment]
Fixed with #198
